File: core/src/main/java/forpleuvoir/dhwu1a/core/message/messageitem/Xml.java
package forpleuvoir.dhwu1a.core.message.messageitem;

import com.google.gson.annotations.SerializedName;
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItem;
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItemType;

import static forpleuvoir.dhwu1a.core.websocket.base.ApiKey.XML;

/**
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.message.messageitem
 * <p>#class_name Xml
 * <p>#create_time 2021/6/30 0:27
 */
public class Xml extends MessageItem {
    /**
     * XML文本
     */
    @SerializedName(XML)
    public final String xml;

    public Xml(String xml) {
        super(MessageItemType.Xml);
        this.xml = xml;
    }

    @Override
    public String toPlainText() {
        return "[XML]";
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/message/messageitem/Xml.kt
package forpleuvoir.dhwu1a.core.message.messageitem

import com.google.gson.annotations.SerializedName
import forpleuvoir.dhwu1a.core.common.XML
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItem
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItemType

/**
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.message.messageitem
 *
 * #class_name Xml
 *
 * #create_time 2021/6/30 0:27
 */
class Xml(
    /**
     * XML文本
     */
    @field:SerializedName(XML) val xml: String
) : MessageItem(MessageItemType.Xml) {
    override fun toPlainText(): String {
        return "[XML]"
    }
}